core/chains/evm/client/node_lifecycle.go

```diff
@@ -151,13 +156,26 @@ func (n *node) aliveLoop() {
 			}
 			if pollFailureThreshold > 0 && pollFailures >= pollFailureThreshold {
```

Comment:
My only concern is that poll interval can be quite long, in some cases significantly longer than the time between heads.

Default poll interval is 10s, but on optimism/arbitrum you might get multiple heads per second. This could lead to long lags/aliasing on the checker.

Is there a reason we can't move it into the select block below, and check on receipt of every new head instead?

Reply:
Other comment got buried:
>I don't think checking on head arrival is sufficient (not often enough) or fair (many in a row is OK, no need to short circuit on the first from a batch). I thought about a separate thread but didn't want to add unnecessary complexity. Are you thinking user facing config? or just an implementation detail?

Reply:
If we could check on every new head across the whole pool, then I think this would make sense, but that is not a mechanism that exists currently. It might thrash too, e.g. if heads arrive in quick bursts from each node. It seems advantageous to have some time threshold in play that stops us from greedily chasing the new best 'instantly' each time when the chain is busier.